packages/variable-resolver/src/browser/common-variable-contribution.ts

```diff
@@ -85,7 +85,7 @@ export class CommonVariableContribution implements VariableContribution {
                 const inputs = !!configuration && 'inputs' in configuration ? configuration.inputs : undefined;
                 const input = Array.isArray(inputs) && inputs.find(item => !!item && item.id === variable);
                 if (!input) {
-                    return undefined;
+                    throw new Error(`Undefined input variable "${variable}" encountered. Remove or define "${variable}" to continue.`);
```

Comment:
@elaihau please revert all changes to resolver

I see what you want now, but we should do it holistically and fix in all other places. Please file an issue for it